File: SDKs/netboxdemo.com/2.4/python/src/sdk/models/operations/dcim_virtual_chassis_read.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class DcimVirtualChassisReadPathParams:
    id: int = field(default=None, metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclass
class DcimVirtualChassisReadRequest:
    path_params: DcimVirtualChassisReadPathParams = field(default=None)
    

@dataclass
class DcimVirtualChassisReadResponse:
    content_type: str = field(default=None)
    status_code: int = field(default=None)
    virtual_chassis: Optional[shared.VirtualChassis] = field(default=None)
    

File: SDKs/netboxdemo.com/2.4/typescript/src/sdk/models/operations/dcimvirtualchassisread.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DcimVirtualChassisReadPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: number;
}


export class DcimVirtualChassisReadRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DcimVirtualChassisReadPathParams;
}


export class DcimVirtualChassisReadResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  virtualChassis?: shared.VirtualChassis;
}
